### PR TITLE
DT-3030: replace NomisUserPersonDetails with new model object

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/DT-3030

### CODE REVIEW

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserRetriesService.kt [hunk 1]
```diff
@@ -10,7 +10,7 @@ import uk.gov.justice.digital.hmpps.oauth2server.auth.repository.UserRepository
 import uk.gov.justice.digital.hmpps.oauth2server.auth.repository.UserRetriesRepository
 import uk.gov.justice.digital.hmpps.oauth2server.azure.AzureUserPersonDetails
 import uk.gov.justice.digital.hmpps.oauth2server.delius.model.DeliusUserPersonDetails
-import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.NomisUserPersonDetails
+import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.NomisApiUserPersonDetails
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserRetriesService.kt:13]: would be good to rename `NomisApiUserPersonDetails` to `NomisUserPersonDetails` as the `Api` is superfluous.

Comment (neilmendum) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserRetriesService.kt]: Good point - addressed

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**